File: agent-driven-architecture-main/examples/typescript/src/spine/ports/event-source.ts
// ── spine/ports/event-source — the sensing seam (raw events in) ─────────────
// INTERFACES ONLY (C11). Perceived content is UNTRUSTED (10.2) and off-bus, so
// it arrives as a `Perceived` that the boundary captures on the committed record
// as one entry of that step's ordered staged fixture (5.4).
//
// The return type is NARROWED to `Perceived`, not `StagedInput`: a sensor cannot
// hand the system a `Recalled`, because a peer tier's conclusion has exactly one
// production site in the system and it is the consumer's bounded relay read.

import type { Perceived } from "../pure/staged";

export interface EventSource {
  poll(): Perceived | null;
}

File: agent-driven-architecture-main/examples/kotlin/spine/src/main/kotlin/adr/spine/ports/EventSource.kt
// ── spine/ports/event-source — the sensing seam ────────────────────────────
// Raw, UNTRUSTED perceived content arriving from the world (10.2). Whatever it
// yields is staged for exactly one turn, captured on that turn's StepRecord, and
// projected into the reasoner's Context — never accumulated.
//
// It yields the NARROW variant. A `Recalled` snapshot has exactly one production
// site in the system — the barge-in consumer, which is the party that bounded the
// read — so no sensing adapter can forge a peer tier's conclusion.

package adr.spine.ports

import adr.spine.pure.StagedInput

interface EventSource {
    fun poll(): StagedInput.Perceived?
}
